### ISSUE TITLE
Build URLs a bit more safely

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/adaptors#578]
This issue affects many adaptors (especially some of our new ones!)

When building URLs, we often use patterns like `www.blah.com/${appId}/${path}`, where `path` is a user defined value.

But should path start with a `/`? Or end with one?

In practice it shouldn't matter, and I think usually it doesn't matter. But we should take care to build well-formed URLs which are robust to leading and trailing slashes from users.

I thought we could use the URL global object for this but on checking the API I'm not so sure. We might be better off using `node:path.resolve(a,b,c)` to build the path

See `commcare` for an example of this.

**Comment by josephjclark** [comment]
This has bitten TD a couple of times in the last couple of days - we really shouldn't be so brittle

**Comment by josephjclark** [comment]
Using undici's baseUrl is a pattern we should use more - undici will deal with at least the base slash. That means adaptors need to use baseUrl rather than assembling their own url. The template needs to do this too

**Comment by josephjclark** [comment]
`path.join` is the answer here:
```
import nodepath from 'node:path'
...
const safePath = nodepath.join(state.configuration.version, resource, path)
```

This will sort out all the leading and trailing slashes for us

**Comment by josephjclark** [comment]
The node path pattern is working really well.

I'm starting to phase this into adaptors and a lot of fixes have been made in #683

This issue was raised as a reminder and tracker. I'm satisfied we have a solution and it's being rolled out as we go. So I'm closing it down.
